hunting/aws/docs/ec2_suspicious_get_user_password_request.md

```diff
@@ -0,0 +1,40 @@
+# EC2 Suspicious Get User Password Request
+
+---
+
+## Metadata
+
+- **Author:** Elastic
+- **Description:** This hunting query identifies when a user makes multiple `GetPasswordData` requests for an EC2 instance. The `GetPasswordData` API call retrieves the encrypted administrator password for an instance running Windows. This API call typically only occurs during the initial launch of an instance or when the password is reset. Multiple requests for the same instance may indicate an adversary attempting to escalate privileges or move laterally within the EC2 environment.
+
+- **UUID:** `408ba5f6-5db7-11ef-a01c-f661ea17fbce`
+- **Integration:** [aws.cloudtrail](https://docs.elastic.co/integrations/aws/cloudtrail)
+- **Language:** `[ES|QL]`
+- **Source File:** [EC2 Suspicious Get User Password Request](../queries/ec2_suspicious_get_user_password_request.toml)
+
+## Query
+
+```sql
+from logs-aws.cloudtrail*
+| where @timestamp > now() - 7 day
+| where event.provider == "ec2.amazonaws.com" and event.action == "GetPasswordData"
+| dissect aws.cloudtrail.request_parameters "{%{?instance_key}=%{instance_id}}"
+| stats instance_count = count(*) by instance_id
```

Comment:
will this work to investigate further who is calling GetPassword API on more than 2 unique instances ? 

`| stats instance_count = count_distinct(instance_id) by aws.cloudtrail.user_identity.arn`

Reply:
Good point, adjusted and tested this. Resulting table will now show the user identity behind these.